src/backend/gl/src/native.rs

```diff
@@ -82,10 +160,17 @@ pub enum ImageView {
 }
 
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
-pub struct DescriptorSetLayout;
+pub(crate) enum DescSetBindings {
+    Buffer(BindingTypes, BindingID, RawBuffer, gl::types::GLintptr, gl::types::GLsizeiptr),
+    Texture(BindingID, Texture),
+    Sampler(BindingID, Sampler),
+}
 
-#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
-pub struct DescriptorSet;
+#[derive(Clone, Debug)]
+pub struct DescriptorSet {
+    layout: Vec<pso::DescriptorSetLayoutBinding>,
+    pub(crate) bindings: Arc<RwLock<Vec<DescSetBindings>>>,
```

Comment:
to recap, if I understand correctly you want `Arc+RwLock` here because this is an internally mutable state that's changed by `write_descriptor_sets`, correct? I believe we could go for simpler `RefCell` here instead, given that Vulkan spec requires that `write_descriptor_sets` doesn't conflict with any descriptor usage (in flight) or another write/copy of descriptor sets.

Reply:
Yes, we could if rust wouldn't error with the following message:
```
error[E0277]: `std::cell::RefCell<std::vec::Vec<native::DescSetBindings>>` cannot be shared between threads safely
  --> src/backend/gl/src/lib.rs:44:6
   |
44 | impl hal::Backend for Backend {
   |      ^^^^^^^^^^^^ `std::cell::RefCell<std::vec::Vec<native::DescSetBindings>>` cannot be shared between threads safely
   |
   = help: within `native::DescriptorSet`, the trait `std::marker::Sync` is not implemented for `std::cell::RefCell<std::vec::Vec<native::DescSetBindings>>`
   = note: required because it appears within the type `native::DescriptorSet`
```

Should I just manually implement the Sync trait?

Reply:
Ah, sorry, you are correct... `RefCell` is `!Sync`. Let's go with `Arc<Mutex>` then, it should be lighter than `RwLock`